Classes/Sitegeist/Monocle/Aspects/FusionCachingAspect.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Sitegeist\Monocle\Aspects;
+
+/**
+ * This file is part of the Sitegeist.Monocle package
+ *
+ * (c) 2016
+ * Martin Ficzel <ficzel@sitegeist.de>
+ * Wilhelm Behncke <behncke@sitegeist.de>
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Aop\JoinPointInterface;
+use Neos\Cache\Frontend\VariableFrontend;
+
+/**
+ * @Flow\Scope("singleton")
+ * @Flow\Aspect
+ */
+class FusionCachingAspect
+{
+    /**
+     * @Flow\Inject
+     * @var VariableFrontend
+     */
+    protected $fusionCache;
+
+    /**
+     * @Flow\Around("setting(Sitegeist.Monocle.fusion.enableObjectTreeCache) && method(Sitegeist\Monocle\Fusion\FusionService->getMergedFusionObjectTreeForSitePackage())")
+     * @param JoinPointInterface $joinPoint The current join point
+     * @return mixed
+     */
+    public function cacheGetMergedFusionObjectTree(JoinPointInterface $joinPoint)
+    {
+        $siteResourcesPackageKey = $joinPoint->getMethodArgument('siteResourcesPackageKey');
+        $cacheIdentifier = str_replace('.', '_', $siteResourcesPackageKey);
```

Comment:
`md5($siteResourcesPackageKey)` might be more stable here because IIRC package keys can contain more characters that are not allowed in cache identifiers (not sure though)

Reply:
This is basically how neos handes the fusion caching itsellf https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Aspects/FusionCachingAspect.php#L37-L38